python_modules/libraries/dagster-aws/dagster_aws/pipes/clients/ecs.py

```diff
@@ -38,7 +42,7 @@ class PipesECSClient(PipesClient, TreatAsResourceParam):
 
     def __init__(
         self,
-        client: Optional[boto3.client] = None,  # pyright: ignore (reportGeneralTypeIssues)
+        client=None,
```

Comment:
Not sure why annotation is removed here but you should explicitly put `Any` if you are changing it to `Any`

Reply:
Because `boto3.client` is an incorrect type annotation. 

`boto3.client("<service>") returns a dynamically generated class, it doesn't return anything like `boto.client`. 

Vanilla `boto3` is untyped (unless `boto3-stubs` is installed) so `Any` is indeed the best option we have here. 

Alternatively, we can do: 

```python
if TYPE_CHECKING:
    # requries boto3-stubs to be installed
    from mypy_boto3_stubs_ecs.client import ECSClient
    
...
    client: Optional["ECSCLient"] = None
```